float64[] s
float64[] a
float64 r
float64[] s_